subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/annotations/InputPropertyAnnotationHandler.java

```diff
@@ -65,9 +67,15 @@ public void validatePropertyMetadata(PropertyMetadata propertyMetadata, TypeVali
         if (File.class.isAssignableFrom(valueType)
             || java.nio.file.Path.class.isAssignableFrom(valueType)
             || FileCollection.class.isAssignableFrom(valueType)) {
-            validationContext.visitPropertyProblem(WARNING,
-                propertyMetadata.getPropertyName(),
-                String.format("has @Input annotation used on property of type '%s'", ModelType.of(valueType).getDisplayName())
+            validationContext.visitPropertyProblem(problem ->
+                problem.withId(ValidationProblemId.INCORRECT_USE_OF_INPUT_ANNOTATION)
+                .forProperty(propertyMetadata.getPropertyName())
+                .reportAs(ERROR)
+                .withDescription(() -> String.format("has @Input annotation used on property of type '%s'", ModelType.of(valueType).getDisplayName()))
+                .happensBecause(() -> "A property of type '" + ModelType.of(valueType).getDisplayName() + "' is annotated with @Input cannot determine how to interpret the file")
+                .addPossibleSolution("Annotate with @InputFile for regular files")
+                .addPossibleSolution("Annotate with @InputDirectory for directories")
```

Comment:
If you want to track the path, you can also return `File.absolutePath`, return a `String` and keep the `@Input`.